randovania/games/am2r/exporter/patch_data_factory.py

```diff
@@ -188,11 +199,15 @@ def _create_starting_location(self) -> dict:
         }
 
     def _create_hash_dict(self) -> dict:
-        return {
+        returnDict = {
             "word_hash": self.description.shareable_word_hash,
             "hash": self.description.shareable_hash,
             "session_uuid": str(self.players_config.get_own_uuid()),
         }
+        starting_memo = self._create_starting_popup(self.patches)
+        if starting_memo:
+            returnDict["starting_memo"] = starting_memo
```

Comment:
That's a type error because your `returnDict` is inferred to be a `dict[str, str]` by the definition in line 202 and following.

Reply:
hmmm how so? the return type for the function is just a dict. i don't have a `dict[str, str]` explictily mentioned anywhere.